docs/documentation.rst

```diff
@@ -0,0 +1,196 @@
+  the code. We name our module at the top with a directive, give a synopsis of
+  this module, provide an author, and then give a much more descriptive summary
+  of what this module does. All this will get formatted by sphinx and added to
+  our webpage.
+* Immediately after the class and function definitions, we put docstrings that
+  explain what this class (or function) does. If you are unsure what to put as
+  a description of your class, have a look at other documentation pages in
+  here and in other projects to get an idea of what would be good summaries.
+* When writing docstrings for functions, good practice is to (at minimum) write
+  descriptions for every parameter and the return value. This can be done (as
+  seen in the example) with the ``:param <parname>:`` and ``:return:`` directives.
+
+We also have an extra function called ``extra_function`` that we might need for
+technical reasons (e.g. for internal calculations, or as boilerplate code to
+interface with other codes elsewhere). Such functions should not need excessive
+comments, but it is good practice to give them clear docstrings either way.
+
+
+Creating a doc page
+-------------------
+
+Documentation in SOLikeT can be very easily created from the docstrings  in your
+python code. For this, SOLikeT uses the **autodoc** extension for sphinx.
+
+Let's have a look at our Pulsars class we created earlier in the
+`Creating Docstrings`_ section.
+
+We start by creating a new, empty text file called ``SOLikeT/docs/pulsars.rst``
+(i.e. the file is called ``pulsars.rst`` and it is stored in the ``docs`` folder
+of SOLikeT). The contents can be quite simple:
+
+.. code::
+
+   Pulsar Theory Code
+   ==================
+   
+   .. automodule:: soliket.pulsars
+   
+   .. autoclass:: soliket.Pulsars
+       :exclude-members: extra_function
+       :members:
+
+At the top we define a title, which is simply **Pulsar Theory Code** (note that
+the underline, which is made with ``=`` symbols, needs to be at least as long
+as the text it's underlining). We then simply add two directives, one
+``automodule`` with the same name as the ``module`` directive at the top of
+our module docstring (see the previous section), and one ``autoclass``
+directive that uses the python name of our class. If you have multiple classes,
+add multiple ``autoclass`` directives, one for each class you have. The
+``:members:`` directive will cause sphinx to turn the docstrings for each of the
+class's member functions to be turned into an entry.
+
+Sometimes, your class might inherit a lot of functions from some parent class,
+or implement a lot of boilerplate functions that are needed to interface with
+some other codes or frameworks. These kinds of functions can at times give
+excessive documentation that the end-user might not really need or should not
+care about. In our Pulsars example, we had the ``extra_function`` that we did
+not want to add to our docs page. For this, there exists the
```

Comment:
While this is useful information, we don't give a directive to make sure people keep this up to date. You could imagine a code that stops being extraneous but remains a ghost in the docpage?